src/cart/utils/crt_utils.c

```diff
@@ -392,9 +397,6 @@ crtu_dc_mgmt_net_cfg_rank_add(const char *name, crt_group_t *group,
 	}
 
 err_group:
-	if (rc != 0) {
```

Comment:
The group structure is used after the call of this function (for example at line 612) , event if this last one return a rc != 0.

Reply:
instead of this being removed, instead calls to this function should fail safely/exit without referencing group if the error was detected during this stage

Reply:
The safely/exit is indirectly done with testing in the caller if the rank list is empty. However, I could explicitly do it just after the call of the function.
From my side, I am not in favor of deallocating memory of the group pointer as it is an input parameter. FMHO, deallocating this memory should be done in the function caller (moreover as this function is exported) as proposed in the patch

Reply:
@frostedcmos , please could you tell me if the proposed modifications are OK for you.
Thanks in advance,
@knard-intel